wrapanapi/systems/openstack.py

```diff
@@ -824,6 +905,13 @@ def get_vm(self, name=None, id=None, ip=None):
             raise MultipleInstancesError('match criteria: {}'.format(kwargs))
         return matches[0]
 
+    def get_instance(self, vm_name_or_id):
+        """Return the virtual machine instance by it's id or name"""
```

Comment:
I think this will be a bit confusing/misleading when `get_vm` exists.  `get_vm` takes advantage of `find_vm`, and the fact that multiple may exist with the same name.  What's the advantage here, just using the connection object to look up instead of using an API client object?

Reply:
The problem here is that get_vm try to get what we may have no access to when listing all tenants and in this case we do not what to filter any thing, we know the exact name of the vm or it's id and  we what to get it's wrapanapi instance (perhaps we can rename it to to get_vm_instance as the original get_vm return a munch) 
here is what we get when trying to get the vm with get_vm function
```python
os_system.get_vm(name="iqe-satellite-test-dlezz-uoaxjidnrg")
Traceback (most recent call last):
  File "/home/dlezz/.pyenv/versions/insights/lib/python3.6/site-packages/IPython/core/interactiveshell.py", line 3291, in run_code
    exec(code_obj, self.user_global_ns, self.user_ns)
  File "<ipython-input-6-9072f2dc6070>", line 1, in <module>
    os_system.get_vm(name="iqe-satellite-test-dlezz-uoaxjidnrg")
  File "/home/dlezz/projects/wrapanapi/wrapanapi/systems/openstack.py", line 901, in get_vm
    matches = self.find_vms(**kwargs)
  File "/home/dlezz/projects/wrapanapi/wrapanapi/systems/openstack.py", line 865, in find_vms
    instances = self.list_vms()
  File "/home/dlezz/projects/wrapanapi/wrapanapi/systems/openstack.py", line 834, in list_vms
    instances = self._generic_paginator(call)
  File "/home/dlezz/projects/wrapanapi/wrapanapi/systems/openstack.py", line 814, in _generic_paginator
    temp_list = f()
  File "/home/dlezz/.pyenv/versions/insights/lib/python3.6/site-packages/novaclient/v2/servers.py", line 861, in list
    "servers")
  File "/home/dlezz/.pyenv/versions/insights/lib/python3.6/site-packages/novaclient/base.py", line 254, in _list
    resp, body = self.api.client.get(url)
  File "/home/dlezz/.pyenv/versions/insights/lib/python3.6/site-packages/keystoneauth1/adapter.py", line 375, in get
    return self.request(url, 'GET', **kwargs)
  File "/home/dlezz/projects/wrapanapi/wrapanapi/systems/openstack.py", line 56, in _request_timeout_handler
    return SessionClient.request(self, url, method, **kwargs)
  File "/home/dlezz/.pyenv/versions/insights/lib/python3.6/site-packages/novaclient/client.py", line 80, in request
    raise exceptions.from_response(resp, body, url, method)
novaclient.exceptions.Forbidden: Policy doesn't allow os_compute_api:servers:detail:get_all_tenants to be performed. (HTTP 403) (Request-ID: req-6d1e9bd3-75e3-4405-b337-c21db450eb75)

```

Reply:
Please also note that I wanted to use the openstack well documented api for getting a concrete  server that is via the connection, in our case getting the connection object is cheap as it's via the already created session.

Reply:
but we can also modify get_vm function to not search for the vm if we do not want to:

```python
def get_vm(self, name=None, id=None, ip=None, search=True):
    # search is True to not break compatibility
    if search is False:
         name_or_id:
              # prefere id or name
              name_or_id = id if id else name
              return self.connection.get_server(name_or_id) 
```
and of course remove that get_instance
what do you think?

Reply:
found an other approach 
```python
    def list_vms(self, filter_tenants=True):
        call = partial(self.api.servers.list, True, {'all_tenants': filter_tenants})

def find_vms(self, name=None, id=None, ip=None, filter_tenants=True):
    ...
    instances = self.list_vms(filter_tenants=filter_tenants)

def get_vm(self, name=None, id=None, ip=None, filter_tenants=True):
    kwargs = {'name': name, 'id': id, 'ip': ip, 'filter_tenants': filter_tenants}
    ...
```

when testing
```python
vm = os_system.get_vm(name="my-rhel8", filter_tenants=False)
vm
Out[5]: <wrapanapi.systems.openstack.OpenstackInstance system=<OpenstackSystem> raw=<novaclient.v2.servers.Server>, kwargs['uuid']='bf7cac9e-2abb-46e0-b7fb-1bb15508afa6'>

```
what do you think?

Reply:
function removed